## Name: Reem Alyahya

---
title: "Project 1: Fast Food Marketing Campaign A/B Test"
---

## Project Overview
  
Evaluating three marketing promotions for a new menu item using A/B testing to determine the most effective strategy based on sales performance.

## Objectives
  
  * Compare sales of three different promotions.
  * Apply statistical analysis to confirm the significance.
  * Decide which promotion campaign is the best to use.


## Data Features
* Promotion: tested marketing campaign.
* SalesInThousands: target sales metric.
* Week: week of the promotin.
* AgeOfStore: store age in years.


---------------------

### import  needed libraries:

In [58]:
# Standard libraries
from pathlib import Path

# Installed libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# Configuration
pd.set_option('display.max_columns', None)
plt.style.use('seaborn-v0_8-darkgrid')
np.random.seed(42)

### Data Loading:
loading campaign data to start the analysis

In [59]:
df = pd.read_csv('WA_Marketing-Campaign.csv')
df

,MarketID,MarketSize,LocationID,AgeOfStore,Promotion,week,SalesInThousands
0,1,Medium,1,4,3,1,33.73
1,1,Medium,1,4,3,2,35.67
2,1,Medium,1,4,3,3,29.03
3,1,Medium,1,4,3,4,39.25
4,1,Medium,2,5,2,1,27.81
...,...,...,...,...,...,...,...
543,10,Large,919,2,1,4,64.34
544,10,Large,920,14,2,1,50.20
545,10,Large,920,14,2,2,45.75
546,10,Large,920,14,2,3,44.29


`Data is successfully loaded with 548 rows and 7 columns that contains [AgeOfStore, Promootion, week, SalesInThousands].`

In [61]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 548 entries, 0 to 547
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   MarketID          548 non-null    int64  
 1   MarketSize        548 non-null    str    
 2   LocationID        548 non-null    int64  
 3   AgeOfStore        548 non-null    int64  
 4   Promotion         548 non-null    int64  
 5   week              548 non-null    int64  
 6   SalesInThousands  548 non-null    float64
dtypes: float64(1), int64(5), str(1)
memory usage: 30.1 KB


### Findings:
* The DataFrame is clean with no missing values.
* All columns have the correct types.

## Confirmatory data analysis (CDA):
In this section, we will calculate the `mean()`, `std()` and `count()` of each promotion group, This allows us to compare their performance.

In [64]:
# Calculate promotions sales average:

df.groupby('Promotion')['SalesInThousands'].agg(['mean', 'std', 'count'])

,mean,std,count
Promotion,,,
1,58.099012,16.553782,172
2,47.329415,15.108955,188
3,55.364468,16.766231,188


### Findings:
* **Promotion1** have the highest sales in average, while the **Promotion2** have the lowest.
* **The Standard deviation** is consistent between groups.
* **The observations of count** are compareable (172 - 188).


   `Now we need to verify if these results are statistically significant `